index.php

```diff
@@ -99,7 +114,7 @@ function do_blocks( $content ) {
 	global $registered_blocks;
 
 	// Extract the blocks from the post content
-	$open_matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*)-->.*<!--\s*\/wp:\g1\s+-->/';
+	$open_matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*)--><!--\s*\/wp:\g1\s+-->/';
```

Comment:
We need the `.*` here, don't we?  Or is this intended to match only blocks with no content?